invenio_pidstore/cli.py

```diff
@@ -0,0 +1,148 @@
+
+from __future__ import absolute_import, print_function
+
+import click
+from flask_cli import with_appcontext
+from invenio_db import db
+
+
+def process_status(ctx, param, value):
+    """Return status value."""
+    from .models import PIDStatus
+
+    # Allow empty status
+    if value is None:
+        return None
+
+    if value not in PIDStatus.attributes:
+        raise click.BadParameter('Status needs to be one of {0}.'.format(
+            ', '.join(PIDStatus.attributes)
+        ))
+    return getattr(PIDStatus, value)
+
+
+#
+# PIDStore management commands
+#
+
+@click.group()
+def pid():
+    """PIDStore management commands."""
+
+
+@pid.command()
+@click.argument('pid_type')
+@click.argument('pid_value')
+@click.option('-s', '--status', default='NEW', callback=process_status)
+@click.option('-t', '--type', 'object_type', default=None)
+@click.option('-i', '--uuid', 'object_uuid', default=None)
+@with_appcontext
+def create(pid_type, pid_value, status, object_type, object_uuid):
+    """Create new persistent identifier."""
+    from .models import PersistentIdentifier
+
+    if bool(object_type) ^ bool(object_uuid):
+        raise click.BadParameter('Speficy both or any of --type and --uuid.')
+
+    new_pid = PersistentIdentifier.create(
+        pid_type,
+        pid_value,
+        status=status,
+        object_type=object_type,
+        object_uuid=object_uuid,
+    )
+    db.session.commit()
+    click.echo(new_pid.id)
+
+
+@pid.command()
+@click.argument('pid_type')
+@click.argument('pid_value')
+@click.option('-t', '--type', 'object_type', default=None)
+@click.option('-i', '--uuid', 'object_uuid', default=None)
+@click.option('--overwrite', is_flag=True, default=False)
+@with_appcontext
+def assign(pid_type, pid_value, object_type, object_uuid, overwrite):
+    """Assign persistent identifier."""
+    from .models import PersistentIdentifier
+
+    if bool(object_type) ^ bool(object_uuid):
+        raise click.BadParameter('Speficy both or any of --type and --uuid.')
+
+    obj = PersistentIdentifier.get(pid_type, pid_value)
+    obj.assign(object_type, object_uuid, overwrite=overwrite)
+    db.session.commit()
+    click.echo(obj.status)
+
+
+@pid.command()
+@click.argument('pid_type')
+@click.argument('pid_value')
+@with_appcontext
+def unassign(pid_type, pid_value):
+    """Unassign persistent identifier."""
+    from .models import PersistentIdentifier
+
+    obj = PersistentIdentifier.get(pid_type, pid_value)
+    obj.unassign()
+    db.session.commit()
+    click.echo(obj.status)
+
+
+@pid.command('object')
+@click.argument('pid_type')
+@click.argument('pid_value')
+@with_appcontext
+def get_object(pid_type, pid_value):
+    """Get an object behind persistent identifier."""
+    from .models import PersistentIdentifier
+
+    obj = PersistentIdentifier.get(pid_type, pid_value)
+    if obj.has_object():
+        click.echo('{0.object_type} {0.object_uuid} {0.status}'.format(obj))
+
+
+@pid.command('links')
```

Comment:
`assign`, `create` and `unassign` are all verbs, while `links` and `object` are not.  It would be nice to use verbs everywhere for consistency.  What about something verbish like `get-pid` and `get-object`?  (or `reference` and `dereference`, or somesuch)